effectful/handlers/llm/sampling.py

```diff
@@ -52,15 +54,48 @@ def sample(template, n):
     def in_nested_call() -> bool:
         return False
 
+    lock = threading.Lock()
```

Comment:
Given a dedicated commit for adding this lock, did you introduce this due to running into a threading issue in an example? It would be good to have this as a test case.

Reply:
Nope. The thought was to make `sample` safer by limiting parallelism to handlers to `completion`.